pkg/types/featuregates/featuregate.go

```diff
@@ -46,12 +44,5 @@ func newFeatureGate(enabled, disabled []configv1.FeatureGateName) FeatureGate {
 // Enabled returns true if the provided feature gate is enabled
 // in the active feature sets.
 func (f *featureGate) Enabled(key configv1.FeatureGateName) bool {
-	if f.enabled.Has(key) {
-		return true
-	}
-	if f.disabled.Has(key) {
-		return false
-	}
-
-	panic(fmt.Errorf("feature %q is not registered in FeatureGates", key))
```

Comment:
Why is this causing issues? This is a deliberate choice within CCO to make sure that there _is_ an opinion on every feature gate. Anything that isn't known, doesn't have a clear distinction of what the value should be in general.

Reply:
Do we need some custom logic in here to categorize the CAPI gate as CustomNoUpgrade?

Reply:
> Why is this causing issues? This is a deliberate choice within CCO to make sure that there is an opinion on every feature gate. Anything that isn't known, doesn't have a clear distinction of what the value should be in general.

Here's how it's causing issues. If we add [a check like this](https://github.com/openshift/installer/pull/7672/files#diff-5d86ca26c9eda26529b273fcdd6f93839ff2c044fa4874a8df8619f2bab88fe4R24) to the code that checks whether **a feature gate that does not belong to a feature set**  is enabled we get a panic. 

The installer only adds featuregates to the custom feature set if they are explicitly set in the install config. So if `FeatureGateClusterAPIInstall` is not specified in the install config, it belongs to no feature set and anytime this code is checked it will panic.

Reply:
> there is an opinion on every feature gate. Anything that isn't known, doesn't have a clear distinction of what the value should be in general.

If every gate belongs to a set this works as is. But if that's not the case, the installer (& myself) is missing how to get an opinion in this scenario.